pkg/util/errors.go

```diff
@@ -76,20 +76,18 @@ func WriteError(ctx context.Context, err error, writer http.ResponseWriter) {
 	}
 }
 
-// HandleResponseError builds at HttpErrorResponse from the given response.
+// HandleResponseError builds an error from the given response
 func HandleResponseError(response *http.Response) error {
 	body, err := BodyToBytes(response.Body)
 	if err != nil {
-		return fmt.Errorf("error processing response body of resp with status code %d: %s", response.StatusCode, err)
+		body = []byte(fmt.Sprintf("error reading response body: %s", err))
 	}
 
 	err = fmt.Errorf("StatusCode: %d Body: %s", response.StatusCode, body)
 	if response.Request != nil {
-		log.C(response.Request.Context()).Errorf("Call to client failed with: %s", err)
-	} else {
-		log.D().Errorf("Call to client failed with: %s", err)
+		return fmt.Errorf("request %s %s failed: %s", response.Request.Method, response.Request.URL, err)
```

Comment:
This will likely propagate information about an internal error (example SM internally calling another system as a client) to the api caller. But it will also make troubleshooting easier. So i am fine either way

Reply:
discussed and as error is not HTTPError, no internal details will be propagated.

Reply:
here a plain error is returned, but only `HTTPError` is sent in the http response
so this should result in status 500 unless it is handled earlier